python/lbann/modules/graph/sparse/GATConv.py

```diff
@@ -0,0 +1,98 @@
+import lbann 
+from lbann.modules import Module 
+from lbann.util import str_list
+from lbann.modules.graph.utils import GraphVertexData
+import lbann.modules
+
+class GatedGraphConv(Module):
+    """Gated Graph Convolution layer. For kernel details, see: 
+
+    https://arxiv.org/abs/1511.05493
+
+    Implementation in the spirit of:
+
+    https://github.com/rusty1s/pytorch_geometric/blob/\
+    master/torch_geometric/nn/conv/gated_graph_conv.py 
+    """
+    global_count = 0
+    def __init__(self, 
+                 output_channels,
+                 num_layers = 1,
+                 name = None,
+                 data_layout = 'data_parallel'):
+        """Initialize GatedGraph layer
+        Args: 
+            output_channels (int): The output size of the node features 
+            num_layers (int): Number of passes through the GRU (default: 1) 
+            name (str): Name of the layers and prefix to use for the layers. 
+            data_layout (str): Data layout (default: data parallel)  
+        """
+        super().__init__()
+
+        ## Add Name for the components for the layer 
+        GatedGraphConv.global_count +=1
+        self.name = (name 
+                    if name 
+                    else 'GatedGraphConv_{}'.format(GatedGraphConv.global_count))
+
+
+        ## Add variables
+        self.output_channels = output_channels
+        self.rnn  = lbann.modules.GRU(output_channels)
+
+        self.num_layers = num_layers
+        self.data_layout = data_layout
+
+        self.weights = [] 
+
+        for i in range(num_layers):
+            weight_init = lbann.Weights(initializer = lbann.NormalInitializer(mean=0, 
+                                                          standard_deviation = 1/((i+1) * output_channels)))
+            weight_layer = lbann.WeightsLayer(dims = str_list([output_channels, output_channels]),
+                                              weights = weight_init, 
+                                              name = self.name+'_'+str(i)+'_weight',
+                                              data_layout = self.data_layout)
+            self.weights.append(weight_layer)
```

Comment:
Don't we only need one `Weights` for this module? We want to apply the same weights at each RNN step.

Reply:
I was a little confused about whether the same weights were applied at each RNN  step or not, and it seemed like the paper was a bit ambiguous. 

Pytorch Geometric seems to use separate weights for each RNN step. https://github.com/rusty1s/pytorch_geometric/blob/df0612bca76b14464b13c86759eec0c56ebc8154/torch_geometric/nn/conv/gated_graph_conv.py#L47-L72

Reply:
Hm, that's a good point. Following PyTorch Geometric's lead is probably the best call then.

In any case, let's also use the initialization in PyTorch, which is uniform in `[-1/sqrt(output_channels),1/sqrt(output_channels)]`.

Reply:
Fixed. Also, that initialization upped the accuracy on test as a bonus.